File: app/src/main/java/org/deiverbum/app/model/OracionSimple.java
package org.deiverbum.app.model;

import static org.deiverbum.app.utils.Utils.LS2;

import android.text.SpannableStringBuilder;

import org.deiverbum.app.utils.ColorUtils;
import org.deiverbum.app.utils.Utils;

public class OracionSimple {
    private String titulo;
    private String texto;
    private String info;

    public OracionSimple() {
    }

    public String getTitulo() {
        return this.titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public String getTexto() {
        return this.texto;
    }

    public void setTexto(String texto) {
        this.texto = texto;
    }

    public String getInfo() {
        return this.info;
    }

    @SuppressWarnings("unused")
    public void setInfo(String info) {
        this.info = info;
    }

    public SpannableStringBuilder getForView(boolean nightMode) {
        ColorUtils.isNightMode = nightMode;

        SpannableStringBuilder sb = new SpannableStringBuilder();
        sb.append(Utils.toH3Red(titulo));
        if (this.info != null) {
            sb.append(LS2);
            sb.append(Utils.toRed(getInfo()));
        }
        sb.append(LS2);
        sb.append(Utils.fromHtml(this.texto));
        return sb;
    }

    public StringBuilder getForRead() {
        StringBuilder sb = new StringBuilder();
        sb.append(getTitulo());
        sb.append(".");
        sb.append(Utils.fromHtml(this.texto));
        return sb;
    }
}

File: app/src/main/java/org/deiverbum/app/model/OracionSimple.kt
package org.deiverbum.app.model

import android.text.SpannableStringBuilder
import org.deiverbum.app.utils.ColorUtils
import org.deiverbum.app.utils.Utils

class OracionSimple {
    var titulo: String? = null
    var texto: String? = null
    private var info: String? = null
    fun getForView(nightMode: Boolean): SpannableStringBuilder {
        ColorUtils.isNightMode = nightMode
        val sb = SpannableStringBuilder()
        sb.append(Utils.toH3Red(titulo))
        if (info != null) {
            sb.append(Utils.LS2)
            sb.append(Utils.toRed(info))
        }
        sb.append(Utils.LS2)
        sb.append(Utils.fromHtml(texto))
        return sb
    }

    val forRead: StringBuilder
        get() {
            val sb = StringBuilder()
            sb.append(titulo)
            sb.append(".")
            sb.append(Utils.fromHtml(texto))
            return sb
        }
}